src/python_testing/TC_WASHERCTRL_2_1.py

```diff
@@ -0,0 +1,125 @@
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+MAX_SPIN_SPEEDS = 16
+
+
+class TC_WASHERCTRL_2_1(MatterBaseTest):
+
+    def desc_TC_WASHERCTRL_2_1(self) -> str:
+        """Returns a description of this test"""
+        return "[TC-WASHERCTRL-2.1] Optional Spin attributes with DUT as Server"
+
+    def pics_TC_WASHERCTRL_2_1(self) -> list[str]:
+        pics = [
+            "WASHERCTRL.S.F00",
+        ]
+        return pics
+
+    def steps_TC_WASHERCTRL_2_1(self) -> list[TestStep]:
+        steps = [
+            TestStep(1, "Commissioning, already done",
+                     is_commissioning=True),
+            TestStep(2, "TH reads from the DUT the SpinSpeeds attribute"),
+            TestStep(3, "TH reads from the DUT the SpinSpeedCurrent attribute"),
+            TestStep(4, "TH writes a supported SpinSpeedCurrent attribute that is a valid index into the list"
+                        + "of spin speeds (0 to numSpinSpeeds - 1)"),
+            TestStep(5, "After a few seconds, TH reads from the DUT the SpinSpeedCurrent attribute"),
+            TestStep(6, "TH writes an unsupported SpinSpeedCurrent attribute that is other than 0 to DUT")
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_WASHERCTRL_2_1(self):
+
+        endpoint = self.get_endpoint(default=1)
+
+        self.step(1)
+
+        # Read the SpinSpeeds attributes
+        self.step(2)
+        list_speed_speeds = await self.read_single_attribute_check_success(endpoint=endpoint,
+                                                                           cluster=Clusters.Objects.LaundryWasherControls,
+                                                                           attribute=Clusters.LaundryWasherControls.Attributes.SpinSpeeds)
+
+        asserts.assert_true(isinstance(list_speed_speeds, list), "Returned value was not a list")
+        numSpinSpeeds = len(list_speed_speeds)
+        asserts.assert_less_equal(numSpinSpeeds, MAX_SPIN_SPEEDS, "List of SpinSpeeds larger than maximum allowed")
+
+        # Read the SpinSpeedCurrent attribute
+        self.step(3)
+        spin_speed_current = await self.read_single_attribute_check_success(endpoint=endpoint,
+                                                                            cluster=Clusters.Objects.LaundryWasherControls,
+                                                                            attribute=Clusters.LaundryWasherControls.Attributes.SpinSpeedCurrent)
+        asserts.assert_true(isinstance(spin_speed_current, int), "SpinSpeedCurrent has an invalid value")
+        asserts.assert_true(0 <= spin_speed_current <= (numSpinSpeeds - 1), "SpinSpeedCurrent outside valid range")
+
+        # Write a valid SpinSpeedCurrent value
+        self.step(4)
+        requested_speed = random.randint(0, numSpinSpeeds - 1)
+        result = await self.default_controller.WriteAttribute(self.dut_node_id,
```

Comment:
suggest using write_single_attribute function - it checks the return status for you and has some nice defaults for node and controller.

https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py#L1244

and below

Reply:
Thanks, I was aware of the read_single_attribute function but didn't knew about this one.

I changed both uses of the WriteAttribute function with the new one.